packages/webview_flutter/webview_flutter_android/android/src/test/java/io/flutter/plugins/webviewflutter/WebViewClientCompatImplTest.java

```diff
@@ -0,0 +1,21 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.webviewflutter;
+
+import android.webkit.WebViewClient;
+
+public class WebViewClientCompatImplTest extends WebViewClientTest {
```

Comment:
Extending a test seems to violate: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#make-each-test-entirely-self-contained. I would prefer just having all the tests added to `WebViewClientTest` with the different classes.

Reply:
Fair enough, I will make the change tomorrow

Reply:
I leaned into the 'Embrace code duplication in tests.' and 'Prefer more test files, avoid long test files'

The test that's independent of the implementation is in `WebViewClientTest` and the tests that should be tested on each implementation are duplicated in `WebViewClientCompatImplTest` and `WebViewClientImplTest`